### PR TITLE
adding Input component

### LINKED ISSUE #7: Input component
## Background
Create a well-defined input component that can adapt to our different needs
that will be used like this
```
<Input
type='email' // password // text // number
lable='my inputs' // string
required // true, false
/>
```

### UAT
- [ ] we can use the custom in our Page one
- [ ] we must pass 8 or more characters for the password field
- [ ] we can pass custom labels to the Input component
- [ ] we can make it required or not through props
- [ ] we can change how it valdates text by passing an input type to it
- [ ] we can view the Page one UI using the Input component
  - [ ] add an extra password field in page one
  - [ ] pass an extra confirm password in page one

### PR DESCRIPTION
**Pull request by ireemalqarni**
closes #7 

Adding Input component designed to accommodate different input types, labels, validation rules, and optional requirements for various forms, including Page1.

- Dynamic Input Types: Easily switch between email, password, text, and number types using the type prop.
Customizable Labels: Set dynamic labels for inputs via the label prop.

- Validation Enhancements:
Enforces a minimum of 8 characters for password fields.
Validation adjusts based on the type prop provided.

- Required Prop: The component can be marked as required or optional through the required prop.

### CODE REVIEW

**Comment by ireemalqarni** [comment]
- Issue with Input Component and Linking Pages
While working on linking multiple pages in my React application, I encountered an issue related to integrating the input form component with other pages. Specifically, the issue arises when attempting to link or render the input component within a larger form structure across different pages.

- What should the code be doing (if it was working)
The input component should be able to capture user input (e.g., email, password, or phone number) and work seamlessly across different pages. It should be rendered properly within the main form and allow users to navigate between pages without disrupting the input functionality.

- What is it doing instead (the problem)
However, when linking the input component with other pages, I am facing errors with input types (in all fields did not taking user input ) and validation problems. Additionally, the input component is either not rendering correctly.

- What is your overall understanding of why it might be happening
I believe the problem could be due to a mismatch in the way the input type is handled (e.g., issues with validation or how certain input types like phone are rendered). It might also be related to how state is managed or how routing and navigation are set up across pages.

Screenshot 2024-10-17 091335

**Comment by ireemalqarni** [comment]
The Page1 component was encountering errors when attempting to pass the name prop to the Input component. This caused form inputs not to update the correct fields in the formData state because the name prop was either missing or not being handled properly in the Input component.

- Changes Made:
Updated Input.tsx:
Added name prop to InputProps interface.
Passed the name prop to the <input> element.

Updated Page1.tsx:
Added name props to all Input fields in the form (name, email, phone, and password).

- proof of success

https://github.com/user-attachments/assets/a1b360ea-d334-4748-8e9b-99b0e4051cc5

File: src/App.css [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
   align-items: center;      /* Center vertically */
 }
 
 .form-container {
-  padding: 300px 200px;
   display: flex;
-  max-width: 300px;
-  max-height: 900px;
+  max-width: 700px !important;
+  max-height: 800px !important;
```
Comment (MaymoonaAlBoloshi) [inline src/App.css]: why do we need  `!important` here

**Review by MaymoonaAlBoloshi — APPROVED** [review]
the styles are not exactly ther, but ti's ok for now

File: src/App.css (hunk 1 above)
Comment (ireemalqarni) [reply src/App.css]: I was use it to force design response

**Pull request merged**